hibernate-reactive-core/src/main/java/org/hibernate/reactive/session/impl/ReactiveStatelessSessionImpl.java

```diff
@@ -222,20 +222,28 @@ public ReactiveEntityPersister getEntityPersister(String entityName, Object obje
     @Override
     public CompletionStage<Void> reactiveUpdate(Object entity) {
         checkOpen();
-        ReactiveEntityPersister persister = getEntityPersister( null, entity );
-        Serializable id = persister.getIdentifier( entity, this );
+        if ( entity instanceof HibernateProxy ) {
+            return reactiveFetch( entity, true ).thenCompose(
+                    fetchedEntity -> executeReactiveUpdate( entity, fetchedEntity ) );
+        }
+        return executeReactiveUpdate( entity, entity );
+    }
+
+    private CompletionStage<Void> executeReactiveUpdate(Object entity, Object updatableEntity) {
```

Comment:
Why do you need both entities as parameters?

Reply:
The **proxy** `entity` is contains in with the updated properties but the **fetched** entity has the non-updated properties. The Object[] state values are pulled from the `entity` but applied to the **fetched** entity. I could pass in the **state** array but the **persister.getPropertyValues(entity)** is used to extract them.

Reply:
The **proxy** `entity` is contains in with the updated properties but the **fetched** entity has the non-updated properties. The Object[] state values are pulled from the `entity` but applied to the **fetched** entity. I could pass in the **state** array but the **persister.getPropertyValues(entity)** is used to extract them.

Reply:
I think it might be a little bit clearer if we renamed the method parameters to `proxy, entity`, or `reference, entity`or something like that.